server/backend/testspg/conftest.py

```diff
@@ -128,3 +131,30 @@ def override_get_rsa_key_method_none():
 def override_get_rsa_key_none(kid):
     return None
 
+
+@pytest.fixture(scope="function")
+def get_current_requester_by_token(db_pg_session):
+    """
+    Convenient fixture to get current requester from token (retrieved from database setup).
+    The fixture returns a function to be called based on access_token's ["username"]
+        , which is the user's cognito_user_id.
+
+    Note, the returned function is an 'async'.
+        To be able to use the returned function from Pytest, please mark the test
+        as '@pytest.mark.asyncio' and use 'async def' for the test with 'await'
+        call to the fixture.
+        (Although it is strange the outer function is not async but it is
+         currently how Pytest can work with async from fixture.)
+    """
+    async def _get_current_requester_by_token(access_token: str) -> Requester:
+
+        claims = jwt.get_unverified_claims(access_token)
+        requester = await get_current_requester(
+            db=db_pg_session,
+            request_cognito_user_id=claims["username"]
```

Comment:
something small, but maybe user the constant "COGNITO_USERNAME_KEY" in `server/backend/api/app/constants.py`?

Reply:
I didn't recall this constant, thanks for mentioning that!!